core/templates/pages/exploration-player-page/services/extract-image-filenames-from-model.service.ts

```diff
@@ -160,8 +163,9 @@ export class ExtractImageFilenamesFromModelService {
 
     _allHtmlInTheState.push(this._getHintsHtml(state));
 
-    if (state.interaction.solution !== null) {
-      _allHtmlInTheState.push(this._getSolutionHtml(state));
+    let solutionHtml = this._getSolutionHtml(state);
+    if (solutionHtml !== '') {
+      _allHtmlInTheState.push(solutionHtml);
```

Comment:
Why is this done?

Reply:
If solutionHtml is not null (which means that the solution in the state does exist), then add solutionHtml to _allHtmlInTheState.